### PR TITLE
Add chengelog entries for bugs which has been fixed by core 0.95.3

### PR DESCRIPTION
**Pull request by zaki50**
please review this @realm/java

### CODE REVIEW

**Comment by kneth** [comment]
Sounds perfect to me. We typically put such entries in a sublist as part of the core upgrade entry to emphasize that it was a core issue.

**Comment by zaki50** [comment]
@kneth thanks. I moved the entry under the core's entry

**Comment by cmelchior** [comment]
But it is still wrong? We need 0.95.2 for it to be truly fixed?

**Comment by zaki50** [comment]
#1803 has been fixed, but not yet #1854.

**Comment by zaki50** [comment]
I'll update this PR after core 0.95.2 will be merged into master.

**Comment by zaki50** [comment]
added a entry for #1854. please review this @realm/java

File: changelog.txt [hunk 1]
```diff
@@ -8,8 +8,10 @@
  * Added new Dynamic API using DynamicRealm and DynamicRealmObject.
  * Added Realm.getSchema() and DynamicRealm.getSchema().
  * Realm.createOrUpdateObjectFromJson() now works correctly if the RealmObject class contains a primary key (#1777).
- * Updated Realm Core to 0.95.1
  * Realm.compactRealm() doesn't throw an exception if the Realm file is opened. It just returns false instead.
+ * Updated Realm Core to 0.95.3
+   - Fixed a bug where RealmQuey.average(String) returns a wrong value for a nullable Long/Integer/Short/Byte field (#1803).
```
Comment (beeender) [inline changelog.txt]: returns -> returned

File: changelog.txt [hunk 2]
```diff
@@ -8,8 +8,10 @@
...
+   - Fixed a bug where RealmQuey.average(String) wrongly counts the null value for average calcuration (#1854).
```
Comment (beeender) [inline changelog.txt]: counts -> counted

File: changelog.txt (hunk 1 above)
Comment (zaki50) [inline changelog.txt]: done

File: changelog.txt (hunk 2 above)
Comment (zaki50) [inline changelog.txt]: done

File: changelog.txt [hunk 3]
```diff
@@ -8,8 +8,10 @@
  * Added new Dynamic API using DynamicRealm and DynamicRealmObject.
  * Added Realm.getSchema() and DynamicRealm.getSchema().
  * Realm.createOrUpdateObjectFromJson() now works correctly if the RealmObject class contains a primary key (#1777).
- * Updated Realm Core to 0.95.1
  * Realm.compactRealm() doesn't throw an exception if the Realm file is opened. It just returns false instead.
+ * Updated Realm Core to 0.95.3
+   - Fixed a bug where RealmQuey.average(String) returned a wrong value for a nullable Long/Integer/Short/Byte field (#1803).
```
Comment (cmelchior) [inline changelog.txt]: average byte field?

File: changelog.txt [hunk 4]
```diff
@@ -8,8 +8,10 @@
...
+   - Fixed a bug where RealmQuey.average(String) wrongly counted the null value for average calcuration (#1854).
```
Comment (Flydiverny) [inline changelog.txt]: calcuration -> calculation :)

File: changelog.txt (hunk 3 above)
Comment (zaki50) [inline changelog.txt]: Oh, we don't support byte!

Comment (zaki50) [inline changelog.txt]: What do you mean?

File: changelog.txt (hunk 4 above)
Comment (zaki50) [inline changelog.txt]: thanks!

File: changelog.txt (hunk 3 above)
Comment (cmelchior) [inline changelog.txt]: Can you take the average of a byte field? I know the underlying representation is INTEGER, it just seems weird?

Comment (zaki50) [inline changelog.txt]: byte is just a 8-bit signed integer in Java, I think

**Comment by cmelchior** [comment]
```
    public double average(String fieldName) {
        long columnIndex = schema.getFieldIndex(fieldName);
        switch (table.getColumnType(columnIndex)) {
            case INTEGER:
                return query.averageInt(columnIndex);
            case DOUBLE:
                return query.averageDouble(columnIndex);
            case FLOAT:
                return query.averageFloat(columnIndex);
            default:
                throw new IllegalArgumentException(String.format(TYPE_MISMATCH, fieldName, "int, float or double"));
        }
    }
```

Apparently we do support it for Byte, just keep it then

File: changelog.txt (hunk 3 above)
Comment (cmelchior) [inline changelog.txt]:
```
    public double average(String fieldName) {
        long columnIndex = schema.getFieldIndex(fieldName);
        switch (table.getColumnType(columnIndex)) {
            case INTEGER:
                return query.averageInt(columnIndex);
            case DOUBLE:
                return query.averageDouble(columnIndex);
            case FLOAT:
                return query.averageFloat(columnIndex);
            default:
                throw new IllegalArgumentException(String.format(TYPE_MISMATCH, fieldName, "int, float or double"));
        }
    }
```

Apparently we do support it for byte, I don't have any sensible use case for it, but no need to change anything in this PR then

Comment (zaki50) [inline changelog.txt]: sure

File: changelog.txt (hunk 4 above)
Comment (zaki50) [inline changelog.txt]: fixed

File: changelog.txt [hunk 5]
```diff
@@ -8,8 +8,10 @@
  * Added new Dynamic API using DynamicRealm and DynamicRealmObject.
  * Added Realm.getSchema() and DynamicRealm.getSchema().
  * Realm.createOrUpdateObjectFromJson() now works correctly if the RealmObject class contains a primary key (#1777).
- * Updated Realm Core to 0.95.1
  * Realm.compactRealm() doesn't throw an exception if the Realm file is opened. It just returns false instead.
+ * Updated Realm Core to 0.95.3
+   - Fixed a bug where RealmQuey.average(String) returned a wrong value for a nullable Long/Integer/Short/Byte field (#1803).
+   - Fixed a bug where RealmQuey.average(String) wrongly counted the null value for average calculation (#1854).
```
Comment (cmelchior) [inline changelog.txt]: Quey -> Query

Comment (zaki50) [inline changelog.txt]: fixed

**Comment by cmelchior** [comment]
:+1:

**Comment by kneth** [comment]
:+1:

**Pull request merged**